### PR TITLE
Bbs 303 user renewal and registration

### PR DESCRIPTION
**Pull request by xendk**

### CODE REVIEW

File: modules/aleph/src/Aleph/AlephClient.php [hunk 1]
```diff
@@ -502,4 +510,107 @@ public function borByKey(AlephPatron $patron) {
...
+    // Tell Aleph that we want to insert a new record.
+    $z305->addChild('record-action', $expiryDate->format('A'));
+    $z305->addChild('z305-id', $patron->getId());
+    // The sub library where memberships are actually registered.
+    $z305->addChild('z305-sub-library', 'BBAAA');
+    // Data needed for creation.
+    $z305->addChild('z305-open-date', $current_date->format('Ymd'));
+    $z305->addChild('z305-update-date', $current_date->format('Ymd'));
+    $z305->addChild('z305-bor-status', '01');
+    $z305->addChild('z305-registration-date', '00000000');
```
Comment (tripox) [inline modules/aleph/src/Aleph/AlephClient.php:580]: Looks like something that should have been the current date? 
However it's Aleph...

File: modules/aleph/src/Aleph/Handler/AlephPatronHandler.php [hunk 2]
```diff
@@ -376,11 +349,93 @@ public function getActiveBranches($bor_id) {
...
+      $this->patron->setExpiryDate($expiryDate);
+    }
+    return $success;
+  }
+
+  /**
+   * Check if update-bor coll was successful.
+   */
+  protected function isUpdateBorSuccess($response) {
+    // Oh yes, the update-bor call reports success with an error message.
```
Comment (tripox) [inline modules/aleph/src/Aleph/Handler/AlephPatronHandler.php:386]: 😭

**Review by tripox — APPROVED** [review]

File: modules/aleph/src/Aleph/AlephClient.php (hunk 1 above)
Comment (xendk) [reply modules/aleph/src/Aleph/AlephClient.php:580]: Yeah, it's "by design". It's been 00000000 for all the users I've seen.

**Pull request merged**